datafusion/expr/src/logical_plan/builder.rs

```diff
@@ -1041,11 +1041,18 @@ pub fn build_join_schema(
     right: &DFSchema,
     join_type: &JoinType,
 ) -> Result<DFSchema> {
+    fn nullify_fields(fields: &[DFField]) -> Vec<DFField> {
```

Comment:
It seems like there are two build_join_schema functions: https://github.com/search?q=repo%3Aapache%2Farrow-datafusion+build_join_schema&type=code

The one in the `datafusion` module already seems to do the right thing with nulls: https://github.com/apache/arrow-datafusion/blob/5b9db17528910328d6ea00d5a6a81138ac6a680b/datafusion/core/src/physical_plan/joins/utils.rs#L349

A nice follow on PR  might be to refactor the code so that only one copy of build_join_schema is present and is used in both places.

Reply:
Thanks @alamb 
This fixes partially #5747 
I will investigate what is the use case for [arrow-datafusion/datafusion/core/src/physical_plan/joins/utils.rs](https://github.com/apache/arrow-datafusion/blob/5b9db17528910328d6ea00d5a6a81138ac6a680b/datafusion/core/src/physical_plan/joins/utils.rs#L349)